### PR TITLE
Update ember-cli and fix deprecations

### PR DESCRIPTION
**Pull request by timmorey**
This updates ember-cli to 3.28 (LTS) via ember-cli-update, and it also resolves all deprecations introduced in the 3.x cycle.

In particular, this addon was triggering the [ember-global](https://deprecations.emberjs.com/v3.x/#toc_ember-global) deprecation, which is blocking us from updating to 4.x.  All we really needed to do to resolve that was update ember-cli-babel.

The diffs are big and diverse, but the commit history walks through the changes fairly discretely.  The vast majority of the changes were just introduced by ember-cli-update, but a few others were manual.

One place I deviated from the default ember blueprints was the introduction of prettier to the project.  This would have re-styled just about every file in the repo, and the current eslint config seems to be working fine.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ module.exports = {
   included(app) {
     this._super.included.apply(this, arguments);
     this.import('vendor/shims/esri-loader.js');
-    this.options = Object.assign({ additionalFiles: [] }, app.options[this.moduleName()])
+    this.addonOptions = Object.assign({ additionalFiles: [] }, app.options[this.moduleName()])
```
Comment (timmorey) [inline index.js:29]: This addresses a build-time deprecation saying

```
DEPRECATION: Ember CLI addons manage their own module transpilation during the `treeForAddon` 
processing. `ember-esri-loader` (found at `...`) has overridden the `this.options.babel` options 
which conflicts with the addons ability to transpile its `addon/` files properly. Falling back to 
default babel configuration options.
```

Looks like I introduced this one with https://github.com/Esri/ember-esri-loader/pull/115 by overwriting `this.options`.

File: tests/dummy/app/routes/application.js [hunk 2]
```diff
@@ -1,17 +1,16 @@
 import { inject as service } from '@ember/service';
 import Route from '@ember/routing/route';
 
-export default Route.extend({
-  esriLoader: service('esri-loader'),
+export default class ApplicationRoute extends Route {
+  @service esriLoader
 
-  renderTemplate: function () {
```
Comment (timmorey) [inline tests/dummy/app/routes/application.js:7]: `renderTemplate` is now deprecated: https://deprecations.emberjs.com/v3.x/#toc_route-render-template

This wasn't causing problems for consuming apps, but it seemed like an easy fix.

File: tests/unit/mixins/esri-module-cache-test.js [hunk 3]
```diff
@@ -1,10 +1,17 @@
+/* eslint-disable ember/no-new-mixins */
+/* eslint-disable ember/no-mixins */
 import EmberObject from '@ember/object';
 import EsriModuleCacheMixin from 'ember-esri-loader/mixins/esri-module-cache';
-import { module } from 'qunit';
-import test from 'ember-sinon-qunit/test-support/test';
+import { module, test } from 'qunit';
+import sinon from 'sinon';
```
Comment (timmorey) [inline tests/unit/mixins/esri-module-cache-test.js:6]: I started getting errors along the way when tests used sinon stuff.  It looks like some of the changes to `qunit`/`ember-qunit` that the updates pulled in broke this.  I tried updating to the latest `ember-sinon-qunit`, but that didn't fix it.  Just using sinon directly seemed easy enough, though it is now our responsibility to explicitly call `sinon.restore()` after tests.

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,11 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## Unreleased
 
+### Breaking
+- Requires node v12+
```
Comment (timmorey) [inline CHANGELOG.md:8]: Recommend releasing as 5.0.0

**Review by tomwayson — APPROVED** [review]
Thanks for this @timmorey!

**Pull request merged**